File: Codeinsight.StreamingManagementSystem/DataAccess/Entities/Subscription.cs
using Codeinsight.StreamingManagementSystem.BusinessLogic.Enums;

namespace Codeinsight.StreamingManagementSystem.DataAccess.Entities
{
    public class Subscription
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public SubscriptionPlanStatus PlanType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public SubscriptionStatus SubscriptionStatus { get; set; }
    }
}

File: authProject/src/app/models/subscription.ts
export interface ISubscription {
  subscriptionId: number;
  userId: number;
  planType: number;
  startDate: string;
  endDate: string;
  subscriptionStatus: number;
}

export interface ISubscriptionRequest {
  planType: number;
  startDate: string;
  endDate: string;
  subscriptionStatus: number;
  userId: number;
}

export interface ISubscriptionDetail {
  message: string;
  subscription: ISubscription[];
}

export interface ICreateSubscriptionResponse {
  message: string;
  subscription: ISubscription;
}
